cylc/flow/scheduler.py

```diff
@@ -1591,11 +1591,13 @@ async def main_loop(self) -> None:
                 ):
                     self.pool.queue_task(itask)
 
+                # Old-style clock-trigger tasks:
                 if (
                     itask.tdef.clocktrigger_offset is not None
                     and itask.is_waiting_clock_done()
+                    # but xtriggers must also be satisfied
+                    and itask.state.xtriggers_all_satisfied()
```

Comment:
Should this be like 1590 just above?:
```suggestion
                    and all(itask.is_ready_to_run())
```

Reply:
Yes I don't see why not (as this still checks the xtriggers)